app/ui-react/packages/api/src/helpers/integrationFunctions.ts

```diff
@@ -686,25 +686,35 @@ export async function setFlow(
  * @param integration
  * @param newFlows
  * @param stepId
- * @param updatedDataShape the input data shape of the conditional flow step to apply as an output data shape to flows
+ * @param flowStartDataShape the input data shape of the conditional flow step to apply as an output data shape to flow start steps
+ * @param flowEndDataShape the output data shape of the conditional flow step to apply as an input data shape to flow end steps
  */
 export function reconcileConditionalFlows(
   integration: Integration,
   newFlows: Flow[],
   stepId: string,
-  updatedDataShape: DataShape
+  flowStartDataShape: DataShape,
+  flowEndDataShape: DataShape
 ) {
   const flowsWithoutStepId = getFlowsWithoutLinkedStepId(
     integration.flows!,
     stepId
   );
   const updatedFlows = newFlows.map(flow => {
-    const newStep = setDataShapeOnStep(
+    const startFlowStep = setDataShapeOnStep(
       { ...flow.steps![0] },
-      updatedDataShape,
+      flowStartDataShape,
       false
     );
-    return applyUpdatedStep(flow, newStep, 0);
+    const updatedFlow = applyUpdatedStep(flow, startFlowStep, 0);
+
+    const endStepPosition = flow.steps!.length-1;
```

Comment:
Smallest nit ever, use [one of these](https://github.com/syndesisio/syndesis/blob/master/app/ui-react/packages/api/src/helpers/integrationFunctions.ts#L946-L966).  I've always thought that at any point we could decide to stop using an array, so I've thought it'd be good to kinda hide that implementation detail.